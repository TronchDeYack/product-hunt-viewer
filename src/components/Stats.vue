<template>
  <div class="stats">
    <div class="stat">
      <div class="number">{{ posts }}</div>
      <div class="label">POSTS</div>
    </div>
    <div class="stat">
      <div class="number">{{ votes }}</div>
      <div class="label">VOTES</div>
    </div>
    <div class="stat">
      <div class="number">{{ comments }}</div>
      <div class="label">COMMENTS</div>
    </div>
    <div class="stat">
      <div class="number">{{ makers }}</div>
      <div class="label">MAKERS</div>
    </div>
  </div>
</template>

<style lang="scss" scoped>
.stats {
  display: inline-flex;
  justify-content: center;
  width: 100%;
  margin: 1rem 0;

  .stat {
    margin: 10px;
    .number {
      color: red;
      font-size: 18px;
    }
    .label {
      font-weight: bold;
      font-size: 12px;
    }
  }
}
</style>

<script>
export default {
  name: 'stats',
  props: {
    posts: Number,
    votes: Number,
    comments: Number,
    makers: Number,
  },
};
</script>
