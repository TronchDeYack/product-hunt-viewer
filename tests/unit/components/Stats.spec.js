import { mount } from '@vue/test-utils';

import Stats from '@/components/Stats.vue';

describe('Stats', () => {
  it('Should render.', () => {
    const wrapper = mount(Stats, {
      propsData: {
        posts: 37,
        votes: 3904,
        comments: 371,
        makers: 54,
      },
    });

    expect(wrapper).toMatchSnapshot();
  });
});
