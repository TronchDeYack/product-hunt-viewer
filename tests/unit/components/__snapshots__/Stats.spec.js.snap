// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Stats Should render. 1`] = `
<div class="stats">
  <div class="stat">
    <div class="number">37</div>
    <div class="label">POSTS</div>
  </div>
  <div class="stat">
    <div class="number">3904</div>
    <div class="label">VOTES</div>
  </div>
  <div class="stat">
    <div class="number">371</div>
    <div class="label">COMMENTS</div>
  </div>
  <div class="stat">
    <div class="number">54</div>
    <div class="label">MAKERS</div>
  </div>
</div>
`;
